src/orchestrator/streaming.rs

```diff
@@ -0,0 +1,326 @@
+        aggregators::{DetectionAggregator, MaxProcessedIndexAggregator},
+        unary::{input_detection_task, tokenize},
+        UNSUITABLE_INPUT_MESSAGE,
+    },
+    pb::{caikit::runtime::chunkers, caikit_data_model::nlp::TokenizationStreamResult},
+};
+
+impl Orchestrator {
+    /// Handles streaming tasks.
+    pub async fn handle_streaming_classification_with_gen(
+        &self,
+        task: StreamingClassificationWithGenTask,
+    ) -> Result<ReceiverStream<ClassifiedGeneratedTextStreamResult>, Error> {
+        info!(
+            request_id = ?task.request_id,
+            model_id = %task.model_id,
+            config = ?task.guardrails_config,
+            "handling streaming task"
+        );
+
+        // TODO:
+        // - Improve error handling for streaming
+        // - Figure out good approach to share generation messages with detection processors (using shared vec for now)
+
+        let ctx = self.ctx.clone();
+        let model_id = task.model_id;
+        let params = task.text_gen_parameters;
+        let input_text = task.inputs;
+
+        // Create response channel
+        let (response_tx, response_rx) = mpsc::channel(1024);
+
+        // Do input detections (unary)
+        let masks = task.guardrails_config.input_masks();
+        let input_detectors = task.guardrails_config.input_detectors();
+        let input_detections = if let Some(detectors) = input_detectors {
+            input_detection_task(&ctx, detectors, input_text.clone(), masks).await?
+        } else {
+            None
+        };
+        debug!(?input_detections);
+        if input_detections.is_some() {
+            // Detected HAP/PII
+            // Do tokenization to get input_token_count
+            let (input_token_count, _tokens) =
+                tokenize(&ctx, model_id.clone(), input_text.clone()).await?;
+            // Send result with input detections
+            let _ = response_tx
+                .send(ClassifiedGeneratedTextStreamResult {
+                    input_token_count,
+                    token_classification_results: TextGenTokenClassificationResults {
+                        input: input_detections,
+                        output: None,
+                    },
+                    warnings: Some(vec![InputWarning {
+                        id: Some(InputWarningReason::UnsuitableInput),
+                        message: Some(UNSUITABLE_INPUT_MESSAGE.to_string()),
+                    }]),
+                    ..Default::default()
+                })
+                .await;
+        } else {
+            // No HAP/PII detected
+            // Do text generation (streaming)
+            let mut generation_stream =
+                generate_stream(&ctx, model_id.clone(), input_text.clone(), params.clone()).await?;
+
+            // Do output detections (streaming)
+            let output_detectors = task.guardrails_config.output_detectors();
+            if let Some(detectors) = output_detectors {
```

Comment:
in case `output_detectors` are empty, it would still be `Some({})` object, so it will get matched here and go into this `if` condition, right? However, we would want it to go to `else` so that we can just process `generation_stream` ? 

Essentially, looks like we may not be following the generation when no detections are requested case ?

Reply:
This is really an incorrect use of `Option`, i.e. it should really be `None` rather than `Some({})` when there are no detectors. I will update it to handle this properly, since it appears we want to do the following rather than `"output": null` or simply omitting the `output` field?

```json
"guardrail_config": {
  "output": {
     "models": {}
  }
}
```

It would probably be good to clean this up during deserialization rather than handling empty nested objects.

Reply:
Pushed fix for this